### PR TITLE
Glt 1280 hibernate notes

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/FormUtils.java [hunk 1]
```diff
@@ -2997,164 +2997,6 @@ public static JSONArray preProcessSampleSheetImport(File inPath, User u, Request
     }
   }
 
-  public static JSONArray processSampleSheetImport(File inPath, User u, RequestManager manager) throws Exception {
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/FormUtils.java]: Have you confirmed with EI that they don't use this?

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java [hunk 2]
```diff
@@ -83,4 +83,9 @@ public void throwIfNotWritable(SecurableByProfile resource) throws IOException,
     return new ArrayList<>(unfiltered);
   }
 
+  @Override
+  public void throwIfNonAdminOrMatchingOwner(User owner) throws IOException, AuthorizationException {
+
+  }
```
Comment (callunity) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java]: Is this method supposed to be blank?

File: core/src/main/java/com/eaglegenomics/simlims/core/Note.java [hunk 3]
```diff
@@ -31,10 +35,19 @@
...
+
+  @Column(nullable = false, updatable = false)
   private Date creationDate = new Date();
+
+  @Column(nullable = false, updatable = false)
   private String text = "";
+
+  @Column(nullable = false, updatable = false)
   @ManyToOne
+  @JoinColumn(name = "owner_userId")
```
Comment (callunity) [inline core/src/main/java/com/eaglegenomics/simlims/core/Note.java]: (You'll need to open the file to see the additional @Column above @ManyToOne

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java (hunk 2 above)
Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java]: Yes, MigrationAuthorizationManager is used to bypass authorization and force use of a configured user when using the MISO Service layer outside of Spring. There is no reason to perform auth checks during migration, and the DefaultAuthorizationManager makes use of a security context, which doesn't exist in migration either

Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java]: This is noted in the class-level Javadoc if you look at the entire file

File: core/src/main/java/com/eaglegenomics/simlims/core/Note.java (hunk 3 above)
Comment (djcooke) [inline core/src/main/java/com/eaglegenomics/simlims/core/Note.java]: True. Will move the properties onto JoinColumn

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java (hunk 2 above)
Comment (callunity) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java]: Yes, I did look at the Javadoc; perhaps it's that I'm still learning the Java conventions as to when it's ok to leave methods blank.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/FormUtils.java (hunk 1 above)
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/FormUtils.java]: Not used from anywhere on develop branch. Confirmed now with @frankie13 that their internal config branch doesn't use it either. I had previously assumed that the "config" branch doesn't do forms processing stuff, but better safe I suppose!

File: migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java (hunk 2 above)
Comment (djcooke) [inline migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MigrationAuthorizationManager.java]: A method body can be blank whenever it has void return type and you don't want it to actually do anything. It's nicer to leave a comment line explaining why it's empty though, so I will do that

**Comment by djcooke** [comment]
Added fixups to address @callunity's comments

**Pull request merged**